cmd/sonobuoy/app/gen.go

```diff
@@ -178,3 +185,53 @@ func getClient(kubeconfig *Kubeconfig) (*kubernetes.Clientset, error) {
 
 	return client, kubeError
 }
+
+// getConfig creates a config with the following algorithm:
+// If no config is supplied defaults will be returned.
+// If a config is supplied then the default values will be merged into the supplied config
+//   in order to allow users to supply a minimal config that will still work.
+// Lastly, options provided on the command line will override
+//   any values in the config.
+func (g *genFlags) getConfig() *config.Config {
```

Comment:
Why move this from sonobuoyconfig.go? If we need a config, then we need to call into the `gen` command now.

Reply:
Because:
 1) it is not used anywhere else
 2) it _requires_ the knowledge of the genFlags, so gen.go seems to be a better place
 3) its the same package so its a pretty minor change

Note: I also unexported it since (1), it isnt being used elsewhere. By default we shouldn't export things w/o a reason.

Reply:
What's interesting that it doesn't break `sonobuoy run`. I would have expected those code paths to use the same data to gen the config that's applied.

Reply:
Yea its just that in most cases people use the default options for these things and the complication of testing CLI flags/input parsing usually means it doesnt happen.